### ISSUE TITLE
Umbraco Forms 8.5.3 - Setting 'StoreUmbracoFormsInDb' to True throws Exception

### ISSUE THREAD

**Issue by jerpenol** [umbraco/Umbraco.Forms.Issues#447]
**Error**
When I tried to set `StoreUmbracoFormsInDb` setting (in `App_Plugins/UmbracoForms/UmbracoForms.config`) to `true`, I received the following exception:
``` Boot failed: Umbraco cannot run. See Umbraco's log file for more details.
-> Umbraco.Core.Exceptions.BootFailedException: Boot failed.
-> Umbraco.Core.Persistence.EntityNotFoundException: Could not find entity IPrevalueSourceEntity with id
  at Umbraco.Forms.Core.Services.PrevalueSourceService.Get(Guid id)
  at Umbraco.Forms.Core.Migrations.V_8_X_X.WorkflowFilesToDb.PrevalueSourceFilesToDb.Migrate()
  at Umbraco.Forms.Core.Migrations.V_8_X_X.AddFormsDbStorage.Migrate()
  at Umbraco.Core.Migrations.MigrationBase.Umbraco.Core.Migrations.IMigration.Migrate() in D:\a\1\s\src\Umbraco.Core\Migrations\MigrationBase.cs:line 73
  at Umbraco.Core.Migrations.MigrationPlan.Execute(IScope scope, String fromState, IMigrationBuilder migrationBuilder, ILogger logger) in D:\a\1\s\src\Umbraco.Core\Migrations\MigrationPlan.cs:line 316
  at Umbraco.Core.Migrations.Upgrade.Upgrader.Execute(IScopeProvider scopeProvider, IMigrationBuilder migrationBuilder, IKeyValueService keyValueService, ILogger logger) in D:\a\1\s\src\Umbraco.Core\Migrations\Upgrade\Upgrader.cs:line 67
  at Umbraco.Forms.Core.Components.UmbracoFormsComponent.Initialize()
  at Umbraco.Core.Composing.ComponentCollection.Initialize() in D:\a\1\s\src\Umbraco.Core\Composing\ComponentCollection.cs:line 32
  at Umbraco.Core.Runtime.CoreRuntime.Boot(IRegister register, DisposableTimer timer) in D:\a\1\s\src\Umbraco.Core\Runtime\CoreRuntime.cs:line 188
```
**Error TLDR;** 
The migration to convert existing Forms (`.json` files) to db-entries throws an exception, because a `PrevalueSource `does not exists in the database.

**Explanation**
This is caused by the migration trying to first get the `PrevalueSource` from the database, since it does not in fact exist (yet) it should be inserted.
The problem here is that the `IBaseService` (which is implemented by the `PrevalueSourceService`) throws an exception since it cannot be found, that in turn prevents the migration from finishing, which results in me getting this error.

I was forced to the delete the json-files, so the migration could proceed.

**Solution**
The solution for this could be to create another method to check if the `PrevalueSource `exists in the database, instead of using the `Get-function` of `IBaseService`.

**Comment by nul800sebastiaan** [comment]
Which version of Forms is this please @jerpenol ?

**Comment by jerpenol** [comment]
@nul800sebastiaan I'm sorry: 8.5.3

**Comment by nul800sebastiaan** [comment]
Thanks! Could you please try one more time in 8.5.5 and see if the error persists? I believe we put some fixes in 8.5.4 for this!

**Comment by nul800sebastiaan** [comment]
I just found this one: https://github.com/umbraco/Umbraco.Forms.Issues/issues/413

Issue should indeed be fixed since 8.5.4 (and 8.5.5 fixes a few more bugs).
